### PR TITLE
Add inputs `matrix-php-versions-unit` and `matrix-php-versions-functional`

### PR DESCRIPTION
**Pull request by BrianHenryIE**
Allows setting the list of PHP versions in the matrix from the caller workflow.

The key is to use `fromJson`:

```yaml
matrix:
  php: ${{ fromJson( inputs.matrix-php-versions-unit ) }}
```

The caller then looks like:

```yaml
jobs:
  test:
    uses: wp-cli/.github/.github/workflows/reusable-testing.yml@main
    with:
      matrix-php-versions-unit: "['7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3']"
      matrix-php-versions-functional: "['7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3']"
```

And the matrix doesn't run PHP 5.6 etc at all:

Screenshot 2023-09-30 at 11 13 37 AM

The PR also removes the `minimum-php` input, since that will be addressed by the PHP matrices.

There will still be an issue with the extra "include" matrix declarations

https://github.com/wp-cli/.github/blob/b0868c02792af659157c03a38254c37e90b099ca/.github/workflows/reusable-testing.yml#L88-L124

[The documentation](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) says:

> Note: All include combinations are processed after exclude. This allows you to use include to add back combinations that were previously excluded.

I wasn't able to use conditional logic / expressions in the matrix `exclude` field.

Any ideas how to handle these combinations? It might be easiest to have all elements in the primary matrices, then exclude what is known to be incompatible combinations, then allow further exclusion to be provided via `input`.

### CODE REVIEW

File: .github/workflows/reusable-testing.yml [hunk 1]
```diff
@@ -3,11 +3,16 @@ name: Testing
...
+        description: 'JSON string array of PHP versions to use in unit test matrix.'
         type: string
         required: false
-        default: '5.6'
+        default: "['5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3']"
+      matrix-php-versions-functional:
+        description: 'JSON string array of PHP versions to use in functional test matrix.'
+        type: string
+        required: false
+        default: "['7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3']"
```
Comment (danielbachhuber) [inline .github/workflows/reusable-testing.yml:15]: Out of curiosity, what happens when this is invalid JSON?

Comment (BrianHenryIE) [reply .github/workflows/reusable-testing.yml:15]: With value `"'7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2', '8.3']"` (I just deleted the leading bracket)
> Error when evaluating 'strategy' for job 'unit'. brianhenryie/bh-wp-github-actions-tests/.github/workflows/reusable-testing.yml@master (Line: 26, Col: 14): Unexpected value '7.1'

**Comment by swissspidy** [comment]
Here's an alternative approach: #78

**Comment by swissspidy** [comment]
Closing in favor of #78

**Pull request closed without merging**